package/kedro_viz/data_access/repositories/modular_pipelines.py

```diff
@@ -179,49 +195,43 @@ def add_child(self, modular_pipeline_id: str, child: ModularPipelineChild):
             >>> assert data_science_pipeline.children == {modular_pipeline_child}
         """
         modular_pipeline = self.get_or_create_modular_pipeline(modular_pipeline_id)
-        modular_pipeline.children.add(child)
-
-    def extract_from_node(self, node: GraphNode) -> Optional[str]:
-        """Extract the namespace from a graph node and add it as a modular pipeline node
-        to the modular pipeline repository.
-
-        Args:
-            node: The GraphNode from which to extract modular pipeline.
-        Returns:
-            ID of the modular pipeline node added to the modular pipeline repository if found.
-        Example:
-            >>> modular_pipelines = ModularPipelinesRepository()
-            >>> model_output_node = GraphNode.create_data_node(
-            ...     "data_science.model_output", layer=None, tags=set(), dataset=None
-            ... )
-            >>> modular_pipelines.extract_from_node(model_output_node)
-            'data_science'
-            >>> assert modular_pipelines.has_modular_pipeline("data_science")
-        """
-
-        # There is no need to extract modular pipeline from parameters
-        # because all valid modular pipelines are encoded in either a TaskNode or DataNode.
-        if isinstance(node, ParametersNode):
-            return None
-
-        modular_pipeline_id = node.namespace
-        if not modular_pipeline_id:
-            return None
+        parent_modular_pipeline_id = modular_pipeline_id.split('.')[0] if '.' in modular_pipeline_id else None

```

Comment:
I do not completely understand this function -

1. If the `modular_pipeline_id` is `a.b.c` then parent of c would be both a and b right ? 
2. If only immediate parent is needed should we do rsplit ? 
3. Should this function just add children (taskNode and dataNode), and may be inputs and outputs for `a.b.c` here ?